modules/linear_boltzmann_solvers/lbs_solver/io/flux_moments_io.cc

```diff
@@ -15,99 +16,71 @@ LBSSolverIO::WriteFluxMoments(
-                            "Header size: " +
-                            std::to_string(num_bytes) +
-                            " bytes\n"
-                            "Structure(type-info):\n"
-                            "uint64_t    num_local_cells\n"
-                            "uint64_t    num_local_nodes\n"
-                            "uint64_t    num_moments\n"
-                            "uint64_t    num_groups\n"
-                            "Each cell:\n"
-                            "  uint64_t    cell_global_id\n"
-                            "  uint64_t    num_cell_nodes\n"
-                            "  Each node:\n"
-                            "    double   x_position\n"
-                            "    double   y_position\n"
-                            "    double   z_position\n"
-                            "Each record:\n"
-                            "  uint64_t    cell_global_id\n"
-                            "  uint64_t    node\n"
-                            "  uint64_t    moment\n"
-                            "  uint64_t    group\n"
-                            "  double      value\n";
-
-  int header_size = (int)header_info.length();
-  char header_bytes[num_bytes];
-  memset(header_bytes, '-', num_bytes);
-  strncpy(header_bytes, header_info.c_str(), std::min(header_size, num_bytes - 1));
-  header_bytes[num_bytes - 1] = '\0';
-
-  file << header_bytes;
-
-  // Write macro data
+  const auto& grid = lbs_solver.Grid();
+  const auto& discretization = lbs_solver.SpatialDiscretization();
+
   const auto& uk_man = lbs_solver.UnknownManager();
   const auto NODES_ONLY = UnknownManager::GetUnitaryUnknownManager();
-  auto& discretization = lbs_solver.SpatialDiscretization();
-  auto& grid = lbs_solver.Grid();
-  const uint64_t num_local_cells = grid.local_cells.size();
-  const uint64_t num_local_nodes = discretization.GetNumLocalDOFs(NODES_ONLY);
-  const uint64_t num_moments = lbs_solver.NumMoments();
-  const uint64_t num_groups = lbs_solver.NumGroups();
-  const auto num_local_dofs = discretization.GetNumLocalDOFs(uk_man);
-  OpenSnLogicalErrorIf(src.size() != num_local_dofs, "Incompatible flux moments vector provided..");
 
-  file.write((char*)&num_local_cells, sizeof(uint64_t));
-  file.write((char*)&num_local_nodes, sizeof(uint64_t));
-  file.write((char*)&num_moments, sizeof(uint64_t));
-  file.write((char*)&num_groups, sizeof(uint64_t));
+  uint64_t num_local_cells = grid.local_cells.size();
+  uint64_t num_local_nodes = discretization.GetNumLocalDOFs(NODES_ONLY);
+  uint64_t num_moments = lbs_solver.NumMoments();
+  uint64_t num_groups = lbs_solver.NumGroups();
+  uint64_t num_local_dofs = num_local_nodes * num_moments * num_groups;
+  OpenSnLogicalErrorIf(src.size() != num_local_dofs, "Incompatible flux moments vector provided.");
 
-  // Write nodal positions
+  H5CreateAttribute(file_id, "num_moments", num_moments);
+  H5CreateAttribute(file_id, "num_groups", num_groups);
```

Comment:
maybe a comment here to state that those attributes are written int he root of the h5 file

Reply:
Added a comment